tensorflow_probability/python/bijectors/ascending.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Ascending bijector."""
+
+import tensorflow.compat.v2 as tf
+
+from tensorflow_probability.python.bijectors import bijector
+from tensorflow_probability.python.internal import assert_util
+
+
+__all__ = [
+    "Ascending",
+]
+
+
+class Ascending(bijector.Bijector):
+  """Maps an unconstrained vector to a vector of (different) elements in
+  ascending order.
+
+  Both the domain and the codomain of the mapping is [-inf, inf], however,
+  the input of the inverse mapping must be strictly increasing.
+
+  On the last dimension of the tensor, the Ascending bijector performs:
+  `y = tf.cumsum([x[0], tf.exp(x[1]), tf.exp(x[2]), ..., tf.exp(x[-1])])`
+
+  #### Example Use:
+
+  ```python
+  bijectors.Ascending().inverse([2, 3, 4])
+  # Result: [2., 0., 0.]
+
+  bijectors.Ascending().forward([0.06428002, -1.07774478, -0.71530371])
+  # Result: [0.06428002, 0.40464228, 0.8936858]
+  ```
+  """
+
+  def __init__(self, validate_args=False, name="ascending"):
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      super().__init__(
+          forward_min_event_ndims=1,
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
+  def _forward(self, x):
+    y0 = x[..., :1]
+    yk = tf.exp(x[..., 1:])
```

Comment:
This means that practically the domain isn't [-inf, inf] but actually just [-709, 709] in float64, and less in float32... larger numbers overfloat the exp() - should this be reflected in the docstring ?